File: src/com/basmilius/ps/bastools/framework/identity/completion/CoreCompletionContributor.java
package com.basmilius.ps.bastools.framework.identity.completion;

import com.intellij.codeInsight.completion.*;
import com.intellij.patterns.PlatformPatterns;
import com.intellij.psi.util.PsiTreeUtil;
import com.intellij.util.ProcessingContext;
import com.jetbrains.php.lang.parser.PhpElementTypes;
import com.jetbrains.php.lang.psi.elements.Method;
import org.jetbrains.annotations.NotNull;

public class CoreCompletionContributor extends CompletionContributor
{

	/**
	 * CoreCompletionContributor Constructor.
	 */
	public CoreCompletionContributor ()
	{
		this.extend(CompletionType.BASIC, PlatformPatterns.psiElement().inside(PlatformPatterns.psiElement(PhpElementTypes.METHOD_REFERENCE)), new PluginCallProvider());
	}

	class PluginCallProvider extends CompletionProvider<CompletionParameters>
	{

		@Override
		protected final void addCompletions (@NotNull final CompletionParameters parameters, final ProcessingContext context, @NotNull final CompletionResultSet results)
		{
			final Method func = PsiTreeUtil.getContextOfType(parameters.getOriginalPosition(), Method.class);

			if (func == null)
				return;
		}

	}

}

File: src/com/basmilius/ps/bastools/framework/identity/completion/CoreCompletionContributor.kt
package com.basmilius.ps.bastools.framework.identity.completion

import com.basmilius.ps.bastools.framework.base.completion.BaseCompletionContributor
import com.basmilius.ps.bastools.framework.base.completion.BaseCompletionProvider
import com.intellij.codeInsight.completion.CompletionParameters
import com.intellij.codeInsight.completion.CompletionResultSet
import com.intellij.codeInsight.completion.CompletionType
import com.intellij.patterns.ElementPattern
import com.intellij.patterns.PlatformPatterns
import com.intellij.psi.PsiElement
import com.intellij.util.PlatformUtils
import com.intellij.util.ProcessingContext

/**
 * Class CoreCompletionContributor
 *
 * @author Bas Milius
 * @package com.basmilius.ps.bastools.framework.identity.completion
 */
class CoreCompletionContributor: BaseCompletionContributor()
{

	/**
	 * CoreCompletionContributor Constructor
	 *
	 * @author Bas Milius
	 */
	init
	{
		this.extend(CompletionType.BASIC, PluginCallProvider())
	}

	/**
	 * {@inheritdoc}
	 *
	 * @author Bas Milius
	 */
	override fun isAvailable(): Boolean
	{
		return PlatformUtils.isPhpStorm()
	}

	/**
	 * Class PluginCallProvider
	 *
	 * @author Bas Milius
	 */
	internal inner class PluginCallProvider: BaseCompletionProvider()
	{

		/**
		 * {@inheritdoc}
		 *
		 * @author Bas Milius
		 */
		override fun isAvailable(): Boolean
		{
			return PlatformUtils.isPhpStorm()
		}

		/**
		 * {@inheritdoc}
		 *
		 * @author Bas Milius
		 */
		override fun addCompletions(parameters: CompletionParameters, context: ProcessingContext, results: CompletionResultSet)
		{
//			val func = PsiTreeUtil.getContextOfType(parameters.originalPosition, Method::class.java) ?: return
		}

		/**
		 * {@inheritdoc}
		 *
		 * @author Bas Milius
		 */
		override fun getPlace(): ElementPattern<out PsiElement>
		{
			return PlatformPatterns.psiElement()
		}

	}

}
